sphinx_gallery/binder.py

```diff
@@ -125,6 +129,10 @@ def check_binder_conf(binder_conf):
         raise ValueError('binder_conf is missing values for: {}'.format(
             missing_values))
 
+    for key in binder_conf.keys():
+        if key not in (req_values + optional_values):
+            raise ValueError("Unknown Binder config key: {}".format(key))
+
```

Comment:
I like this check on allowed configuration keys. We should do this for the rest of the configuration options in another PR